client/mysqladmin.cc

```diff
@@ -373,6 +373,11 @@ int main(int argc,char *argv[])
   int error= 0, temp_argc;
   MYSQL mysql;
   char **commands, **save_argv, **temp_argv;
+  char *tmp;
+  
+  tmp= (char *) getenv("MYSQL_HOST");
```

Comment:
As this is new functionally, lets start with with `MARIADB_HOST` and add the same compatibility to the mysql.cc (doesn't have to be this PR).

Reply:
I'm unsure why this is casting using `(char *)`. I looks like getenv returns a `char *`.